app/assets/javascripts/govuk-component/tasklist.js

```diff
@@ -0,0 +1,333 @@
+// Most of this is taken from the service manual
```

Comment:
Does this JS differ in any way to the accordion component?
Are the differences only in the class names of elements?
If so, can those class names be abstracted somehow?
Is it appropriate to avoid duplication of this JS?

Reply:
It is different (details below) and I think it's going to continue to get more different. In terms of avoiding duplication or building upon the existing accordion, that might be possible except we'd have to separate out the tracking, which is fairly heavily intertwined (the accordion tracks some content clicks based on an assumed content, which isn't ideal - I'm learning more about this all the time).

Differences are:

- 'remember last section opened in URL hash' functionality disabled by default
- new functionality to allow a section to be opened by default

I'm recommending that we remove the first item altogether as it's not needed when the its in the sidebar and having it on the main one only would be inconsistent.

Also, coming soon are some changes to how the GA tracking works.

Reply:
I might update that comment to be clearer..

Reply:
The first point
> 'remember last section opened in URL hash' functionality disabled by default

That feature is still needed for users who browse from a task list to a content page and then hit "Back", the section should remain open so they can continue from where they left off.

Reply:
Two thoughts about that.

Yes, in an ideal world. But the solution doesn't remember all the accordion sections that were opened, only the last one - so the page might look different when they click 'back' anyway.

Behavioural consistency is important, and we're not planning on having this when it's in the sidebar.